plugins/inputs/mac_wireless/mac_wireless.go

```diff
@@ -0,0 +1,132 @@
+package mac_wireless
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"os"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+// default file paths
+const (
+	CMD = "/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport -I"
+)
+
+type Wireless struct {
+	CMD       string `toml:"cmd"`
+	DumpZeros bool   `toml:"dump_zeros"`
+}
+
+var sampleConfig = `
+  ## command to get wireless info. If empty default will be used:
+  ##  
+  ## This can also be overridden with env variable, see README.
+  cmd = "/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport -I"
+  ## dump metrics with 0 values too
+  dump_zeros       = true
+`
+
+func (ns *Wireless) Description() string {
+	return "Collect wireless interface link quality metrics"
+}
+
+func (ns *Wireless) SampleConfig() string {
+	return sampleConfig
+}
+func exe_cmd(cmd string, wg *sync.WaitGroup) ([]byte, error) {
+	parts := strings.Fields(cmd)
+	head := parts[0]
+	parts = parts[1:len(parts)]
+	out, err := exec.Command(head, parts...).Output()
+	wg.Done() // Need to signal to waitgroup that this goroutine is done
+	return out, err
+}
+
+func (ns *Wireless) Gather(acc telegraf.Accumulator) error {
+	// load paths, get from env if config values are empty
+	ns.loadPath()
+
+	// collect wireless data
+	wg := new(sync.WaitGroup)
+	wg.Add(3)
+	wireless, err := exe_cmd(CMD, wg)
+	if err != nil {
+		return err
+	}
+	err = ns.gatherWireless(wireless, acc)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ns *Wireless) gatherWireless(data []byte, acc telegraf.Accumulator) error {
+	metrics, err := loadWirelessTable(data, ns.DumpZeros)
+	if err != nil {
+		return err
+	}
+	tags := map[string]string{
+		"name": "airport",
+	}
+	acc.AddFields("wireless", metrics, tags)
+	return nil
+}
+
+func loadWirelessTable(table []byte, dumpZeros bool) (map[string]interface{}, error) {
+	metrics := map[string]interface{}{}
+	myLines := strings.Split(string(table), "\n")
+	for x := 0; x < len(myLines)-1; x++ {
+		f := strings.Split(myLines[x], ":")
+		f[0] = strings.Trim(f[0], " ")
+		f[1] = strings.Trim(f[1], " ")
+		n, err := strconv.ParseInt(strings.Trim(f[1], " "), 10, 64)
+		if err != nil {
+			continue
```

Comment:
maybe if it isn't an int, it could be added as a tag. stuff like ssid might be useful. declare new tags as
`tags := make(map[string]string)` then add `tags[f[0]] = f[1]` and update the function to return it